src/components/Layout/Header/Header.tsx

```diff
@@ -1,8 +1,78 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
+import { NavLink } from 'react-router-dom'
 import styles from './Header.module.css'
 
-function Header() {
-  return <div className={styles.root}>Header</div>
+interface HeaderProps {
+  logo: string
+  navItems: string[]
+  onNavItemClick: (item: string) => void
+  activeItem: string
+  bgColor?: string
+  textColor?: string
+}
+
+function Header({
+  logo,
+  navItems,
+  onNavItemClick,
+  activeItem,
+  bgColor,
+  textColor
+}: HeaderProps) {
+  const [isNavOpen, setIsNavOpen] = useState(false)
+
+  const openHamburger = () => {
+    setIsNavOpen(!isNavOpen)
+  }
+
+  useEffect(() => {
+    if (document.body.style.overflow === 'hidden') {
+      document.body.style.overflow = 'auto'
+    } else {
+      document.body.style.overflow = 'hidden'
+    }
+  }, [isNavOpen])
+
+  return (
+    <div className={styles.root} style={{ backgroundColor: bgColor, color: textColor }}>
+      <div className={styles.headerContainer}>
+        <div className={styles.logo_container}>
```

Comment:
This is my fault, let's not have snake and camel casing in the same project 

❌
```
      <div className={styles.headerContainer}>
        <div className={styles.logo_container}>
```

✅
```
      <div className={styles.header_container}>
        <div className={styles.logo_container}>
```